java-collections/src/main/java/dev/pawel/linkedlist/MyLinkedList.java

```diff
@@ -0,0 +1,144 @@
+package dev.pawel.linkedlist;
+
+import dev.pawel.MyList;
+
+public class MyLinkedList<E> implements MyList<E> {
+    private final static int DEFAULT_SIZE = 0;
+    private Node<E> head;
+    private int size;
+
+    public MyLinkedList() {
+        this.head = null;
+        size = DEFAULT_SIZE;
+    }
+
+    @Override
+    public E get(int index) {
+        Node<E> node = getNodeByIndex(index);
+        return node.data;
+    }
+
+    @Override
+    public E set(int index, E element) {
+        Node<E> setElement = getNodeByIndex(index);
+        setElement.data = element;
+        return setElement.data;
+    }
+
+    @Override
+    public boolean add(E element) {
+        if (head == null)
+            head = new Node<>(element);
+        else {
+            Node<E> temp = getLastNode();
+            temp.next = new Node<>(element);
+        }
+        size++;
+        return true;
+    }
+
+    @Override
+    public void add(int index, E element) {
+        Node<E> addNode = new Node<>(element);
+        Node<E> add = changeBondsBetweenNodes(addNode, getNodeByIndex(index - 1));
+        add.next = addNode;
+        size++;
+    }
+
+    @Override
+    public E remove(int index) {
+        Node<E> removeNode = getNodeByIndex(index);
+        remove(removeNode.data);
+        return removeNode.data;
+    }
+
+    @Override
+    public boolean remove(E element) {
+        Node<E> removeNode =
+                changeBondsBetweenNodes(getNodeByIndex(indexOf(element) - 1), getNodeByIndex(indexOf(element)));
+        removeNode.next = null;
+        size--;
+        return true;
+    }
+
+    @Override
+    public int size() {
+        return size;
+    }
+
+    @Override
+    public boolean contains(E element) {
+        Node<E> temp = getLastNode();
```

Comment:
I think it won't work for elements that are not the last node; add a test to check if the LinkedList contains a first/middle element